jerry-core/jerryscript.h

```diff
@@ -322,9 +329,19 @@ jerry_value_t jerry_get_object_keys (const jerry_value_t obj_val);
 jerry_value_t jerry_get_prototype (const jerry_value_t obj_val);
 jerry_value_t jerry_set_prototype (const jerry_value_t obj_val, const jerry_value_t proto_obj_val);
 
+__attribute__((deprecated))
```

Comment:
I have no good suggestion. Imagine a situation where a GCC compiled JerryScript library is linked by a TI compiler. This is possible because of the common binary object format. The jerryscript.h is not depend on jrt.h, so the project using the TI compiler can use it. However, this use case would not be possible anymore.

I see two options: remove it, or add a define somewhere at the beginning, which defines a macro depending on the compiler. Perhaps having this define would be useful later when more functions will become deprecated.